gstools/transform/field.py

```diff
@@ -67,6 +69,56 @@ def binary(fld, divide=None, upper=None, lower=None):
         fld.field[fld.field <= divide] = lower
 
 
+def discrete(fld, values, thresholds=None):
+    """
+    Discrete transformation.
+
+    After this transformation, the field has only `len(values)` discrete
+    values.
+
+    Parameters
+    ----------
+    fld : :any:`Field`
+        Spatial Random Field class containing a generated field.
+        Field will be transformed inplace.
+    values : :any:`np.ndarray`
+        The discrete values the field will take
+    thresholds : :class:`numpy.ndarray`, optional
+        the thresholds, where the value classes are separated
+        Default: mean of the neighbouring values
+    """
+    if fld.field is None:
+        print("discrete: no field stored in SRF class.")
+    else:
+        if thresholds is None:
+            # just in case, sort the values
+            values = np.sort(values)
+            thresholds = (values[1:] + values[:-1]) / 2
+        else:
+            if len(values) != len(thresholds) + 1:
+                raise ValueError(
+                    "discrete transformation: len(values) != len(thresholds) + 1"
+                )
+            values = np.array(values)
+            thresholds = np.array(thresholds)
+            for i in range(len(thresholds)):
+                if not (values[i] <= thresholds[i] < values[i + 1]):
```

Comment:
Why this restriction?

Reply:
How else would be define an array of thresholds, which subdivide an array of values? - It's a bit like the nodes and the edges of a graph.

Reply:
The `binary` transformation does the following:
1. get a `divide` value to select "lower" and "upper" values
2. replace the lower and upper values with given values

The divide values is unrelated to the given lower and upper values, which will be set in the field. It is only related to the input-field